tla-assignments/src/main/scala/at/forsyte/apalache/tla/assignments/SymbTransGenerator.scala

```diff
@@ -247,12 +255,32 @@ class SymbTransGenerator( tracker : TransformationTracker ) {
                         strategy : StrategyType,
                         selections : SelMapType
                       ) : Seq[SymbTrans] =
-      selections( ex.ID ).map( s =>
-        (
-          mkOrdered( s, strategy ),
-          sliceWith( s, selections )( ex )
-        )
-      ).toSeq
+      selections( ex.ID ).map { s =>
```

Comment:
This block is pretty hairy. Do there's an relatively way way of refactoring this a bit so that the reader doesn't have to grok so much activity within this lambda? Perhaps by breaking it into a few named functions (for the sake of self-documentation) or at least one named function defined prior to actually calling the function on the selections?

Reply:
If you want, sure, should be easy enough to split.

Reply:
I know it would help with readability for me. I suspect it'd help future readers too.